src/common/snippets/src/lowered/expression_factory.cpp

```diff
@@ -129,7 +129,9 @@ ExpressionPtr LinearIR::ExpressionFactory::create(const std::shared_ptr<ov::Node
     expr->validate();
     // todo: here we blindly synchronize input shapes from parent and child. Remove this when shapes will be stored in
     //  port connector itself
-    if (linear_ir.m_shape_infer_factory)
+    if (linear_ir.m_shape_infer_factory &&
+        !ov::is_type<op::PerfCountBeginBase>(n) &&
+        !ov::is_type<op::PerfCountEndBase>(n))
```

Comment:
ShapeInfer for `PerfCount*` ops is trivial and almost free in terms of performance, so do we really need this check?

Reply:
make sense. it's even have better performance w/o check as in release there is no perf count node.